### PR TITLE
Expand installation documentation

### LINKED ISSUE #4: Installation Instructions are incomplete
## Description
The Documentation in the README.md lacks a fully fleshed out install section, currently it skips over the need for PHP, Composer, and Node. It would also be helpful to include Issue and PR templates.

## Additional Notes
The instructions to be added should have sub-sections that can describe the different steps for MacOS as well as *nix systems / WSL2.

### PR DESCRIPTION
**Pull request by m-triassi**
## Description
Closes #4 

Adds nearly all steps needed to set up / install environment related dependencies for the project.  This includes instructions for most possible platforms that developers on the project might be using. Specifically, MacOS, and Windows/WSL

## Testing

These instructions should probably be tested  by someone who hasn't set up the project yet to ensure that they are accurate and cover most cases. Follow all of the relevant steps and report any problems as change requests on this PR

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,14 +16,107 @@ Concordia University
...
+```
+
+Once it completes, you can continue with the installation instructions.
+
+### PHP
+First you'll need PHP running on your command line. To do this install it, as well as some of its dependancies.
+
+For MacOS run: 
+```shell
+brew install php@7.2
```
Comment (willyyhuang) [inline README.md]: i had issue with composer when php is < @7.2, is it recommended to download @^8.0?

Comment (m-triassi) [reply README.md]: Ah yeah you're right, that's a typo, good catch, it should be 7.4. Fixed!

**Review by RobertoNittolo — COMMENTED** [review]
```
### Node
You'll also need node to build the front end, as  well as mange the dependencies there as well. It is highly recommended 
that you use [NVM](https://github.com/nvm-sh/nvm) that way you can manage different versions of node going forward.  

On either WSL, Ubuntu, MacOS, run the following commands:
```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash
nvm install 14
nvm alias default 14
```
Initially was getting the error that nvm was unable to be found, restarting the terminal allowed for these commands to work.

**Comment by RobertoNittolo** [comment]
Was able to successfully get the project running on windows 10 pro version 10.0.19042 build 19042. The instructions were great! Good Job!

**Comment by m-triassi** [comment]
Noticed a missing step, so i added them, (`composer install`, `yarn install`, and `yarn dev`). Otherwise Thanks for trying them, Merging!

**Pull request merged**